File: sgpi.client/src/app/interfaces/item-catalogo.ts
import { Categoria } from './categoria';

export interface ItemCatalogo {
  id: number;
  nome: string;
  descricaoDetalhada: string;
  unidadeMedida: UnidadeMedida;
  categoriaId: number;
  categoria?: Categoria;
}

export enum UnidadeMedida {
  UN = 0,
  KIT = 1,
  METRO = 2,
}

File: Sgpi.Server/Core/Entities/ItemCatalogo.cs
namespace SGPI.Core.Entities
{
    public class ItemCatalogo
    {
        public int Id { get; set; }
        public string Nome { get; set; } = string.Empty;
        public string DescricaoDetalhada { get; set; } = string.Empty;
        public UnidadeMedida UnidadeMedida { get; set; }
        public int CategoriaId { get; set; }
        public Categoria? Categoria { get; set; }
    }

    public enum UnidadeMedida
    {
        UN,
        KIT,
        METRO
    }
}
